carp.protocols.core/src/commonMain/kotlin/dk/cachet/carp/protocols/domain/data/IntervalSamplingConfiguration.kt

```diff
@@ -0,0 +1,22 @@
+package dk.cachet.carp.protocols.domain.data
+
+import dk.cachet.carp.common.TimeSpan
+import dk.cachet.carp.protocols.domain.devices.DeviceDescriptor
+import kotlinx.serialization.Serializable
+
+
+/**
+ * A sampling configuration which allows configuring the time [interval] in between subsequent measurements.
+ */
+@Serializable
+data class IntervalSamplingConfiguration( val interval: TimeSpan ) : SamplingConfiguration()
```

Comment:
The period vs frequency discussion again (see comment in design doc). By using a frequency in Hz you'll also allow the client to do related time calculations however it see fit.

Reply:
Isn't interval <-> frequency a two-way calculation? I don't see that as an argument for or against using one or the other. What matters is for the data type what the most expressive metric is. I argue for stepcount it is definitely `interval` (or `duration`, `period`, ... whichever we agree on).